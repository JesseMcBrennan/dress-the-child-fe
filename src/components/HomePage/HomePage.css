.main-logo {
  font-family: 'dtc-font', sans-serif;
  color: #047177;
  font-size: 4rem;
  content: url('../../images/DressTheChild_FinalLogo.svg');
  height: 11rem;
  margin: 3rem auto;
}

.home-title {
  font-family: 'helvetica', sans-serif;
  font-weight: 550;
  font-size: 2.2rem;
  color: whitesmoke;
  margin: 20rem auto 3rem auto;
  padding: 0 2rem;
}

.donate-btn {
  color: #047177;
  padding: 0.5rem 1.5rem;
  border: 0.1rem solid #047177;
  font-size: 2rem;
  text-decoration: none;
  border-radius: 0.15rem;
  cursor: pointer;
  background: whitesmoke;
  transition: 0.5s;
}

.donate-btn:hover {
  color: whitesmoke;
  border: 0.1rem solid whitesmoke;
  background: #047177;
}

.donate-btn:active {
  color: #047177;
  border: 0.1rem solid #047177;
  background: whitesmoke;
}

@media only screen and (max-width: 610px) {
  .home-title {
    font-size: 2rem;
  }
}
@media only screen and (max-width: 560px) {
  .home-title {
    font-size: 1.8rem;
  }
}
@media only screen and (max-width: 508px) {
  .home-title {
    font-size: 1.6rem;
  }
}
