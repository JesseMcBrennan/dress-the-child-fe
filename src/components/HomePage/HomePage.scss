$teal: #047177;

.main-logo {
  font-family: 'dtc-font', sans-serif;
  color: $teal;
  font-size: 4rem;
  content: url('../../images/DressTheChild_FinalLogo.svg');
  height: 11rem;
  margin: 3rem auto;
}

.home-title {
  font-family: 'helvetica', sans-serif;
  font-weight: 550;
  font-size: 2.2rem;
  color: whitesmoke;
  margin: 20rem auto 3rem auto;
}

.donate-btn {
  color: $teal;
  padding: 0.5rem 1.5rem;
  border: 0.1rem solid $teal;
  font-size: 2rem;
  text-decoration: none;
  border-radius: 0.15rem;
  cursor: pointer;
  background: whitesmoke;
  transition: 0.5s;
}

.donate-btn:hover {
  color: whitesmoke;
  border: 0.1rem solid whitesmoke;
  background: $teal;
}

.donate-btn:active {
  color: $teal;
  border: 0.1rem solid $teal;
  background: whitesmoke;
}
