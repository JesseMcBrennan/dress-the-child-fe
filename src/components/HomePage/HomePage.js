import React from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import './HomePage.css';

export const HomePage = () => (
  <div className="home-card">
    <h1 className="home-title">
      PROVIDING CHILDREN IN NEED
      <br />
      THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING
    </h1>
    <NavLink exact to="/donate" className="donate-btn">
      Donate
    </NavLink>
  </div>
);

export default withRouter(HomePage);
