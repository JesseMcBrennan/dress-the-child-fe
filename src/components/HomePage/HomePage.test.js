import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from './HomePage';

describe('HomePage', () => {
  let mockLocation;
  let wrapper;
  beforeEach(() => {
    mockLocation = {
      pathname: '/'
    };
    wrapper = shallow(<HomePage location={mockLocation} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
