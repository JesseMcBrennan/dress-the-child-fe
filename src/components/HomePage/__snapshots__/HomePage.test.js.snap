// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    location={
      Object {
        "pathname": "/",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="home-title"
        >
          PROVIDING CHILDREN IN NEED
          <br />
          THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING
        </h1>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="donate-btn"
          exact={true}
          to="/donate"
        >
          Donate
        </NavLink>,
      ],
      "className": "home-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "PROVIDING CHILDREN IN NEED",
            <br />,
            "THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING",
          ],
          "className": "home-title",
        },
        "ref": null,
        "rendered": Array [
          "PROVIDING CHILDREN IN NEED",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Donate",
          "className": "donate-btn",
          "exact": true,
          "to": "/donate",
        },
        "ref": null,
        "rendered": "Donate",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="home-title"
          >
            PROVIDING CHILDREN IN NEED
            <br />
            THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING
          </h1>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="donate-btn"
            exact={true}
            to="/donate"
          >
            Donate
          </NavLink>,
        ],
        "className": "home-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "PROVIDING CHILDREN IN NEED",
              <br />,
              "THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING",
            ],
            "className": "home-title",
          },
          "ref": null,
          "rendered": Array [
            "PROVIDING CHILDREN IN NEED",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "THE OPPORTUNITY TO SHOP FOR NEW WINTER CLOTHING",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Donate",
            "className": "donate-btn",
            "exact": true,
            "to": "/donate",
          },
          "ref": null,
          "rendered": "Donate",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
